server/base/src/main/java/org/apache/accumulo/server/ServerContext.java

```diff
@@ -445,4 +462,109 @@ public AuditedSecurityOperation getSecurityOperation() {
     return securityOperation.get();
   }
 
+  public void validateSpiConfiguration(boolean validateNsAndTables)
+      throws AccumuloException, AccumuloSecurityException {
+    boolean valid = validateClasses(getSiteConfiguration());
+    valid = valid && validateClasses(getConfiguration());
+    if (validateNsAndTables) {
+      for (String ns : namespaceOperations().list()) {
+        NamespaceId nsId = NamespaceId.of(namespaceOperations().namespaceIdMap().get(ns));
+        valid = valid && validateClasses(getNamespaceConfiguration(nsId));
+      }
+      for (String table : tableOperations().list()) {
+        TableId tableId = TableId.of(tableOperations().tableIdMap().get(table));
+        valid = valid && validateClasses(getTableConfiguration(tableId));
+      }
+    }
+    if (!valid) {
+      throw new IllegalStateException("SPI class configuration validation failed.");
+    }
+  }
+
+  private boolean validateClasses(AccumuloConfiguration conf) {
+
+    boolean valid = true;
+    for (Property p : Property.values()) {
```

Comment:
This loop will not pick up iterators and compaction services as those classes fall under a prefix and not a specific property.